Sources/SwifterAuth.swift

```diff
@@ -114,6 +115,63 @@ public extension Swifter {
+                      failure: FailureHandler? = nil) {
+        guard let client = client as? SwifterAppProtocol else {
+            let error = SwifterError(message: "SSO not supported AppOnly client",
+                                     kind: .invalidClient)
+            failure?(error)
+            return
+        }
+        
+        let urlScheme = "swifter-\(client.consumerKey)"
+        
+        let nc = NotificationCenter.default
+        self.swifterCallbackToken = nc.addObserver(forName: .swifterSSOCallback, object: nil, queue: .main) { notification in
+            self.swifterCallbackToken = nil
+            let url = notification.userInfo![CallbackNotification.optionsURLKey] as! URL
+            guard url.scheme == urlScheme else { return }
+            
+            let isCanceled = url.host == nil
+            if isCanceled {
+                let error = SwifterError(message: "User cancelled login from Twitter App",
+                                         kind: .cancelled)
+                failure?(error)
+            } else {
+                let params = url.queryParamsForSSO
+                let secret = params["secret"]!
+                let token = params["token"]!
+                let credentialToken = Credential.OAuthAccessToken(key: token, secret: secret)
+                self.client.credential = Credential(accessToken: credentialToken)
+                success?(credentialToken)
+            }
+        }
+        
+        let url = URL(string: "twitterauth://authorize?consumer_key=\(client.consumerKey)&consumer_secret=\(client.consumerSecret)&oauth_callback=\(urlScheme)")!
+        UIApplication.shared.open(url, options: [:], completionHandler: { (success) in
+            if !success {
+                let error = SwifterError(message: "Cannot open twitter app",
+                                         kind: .noTwitterApp)
+                failure?(error)
+            }
+        })
+    }
+    
+    class func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any]) -> Bool {
+        guard let sourceApplication = options[.sourceApplication] as? String else { return false }
+        let isSSO = sourceApplication.hasPrefix("com.twitter") || sourceApplication.hasPrefix("com.atebits")
+        let isWeb = sourceApplication.hasPrefix("com.apple") || sourceApplication == Bundle.main.bundleIdentifier!
+        if isSSO {
+            let notification = Notification(name: .swifterSSOCallback, object: nil, userInfo: [CallbackNotification.optionsURLKey: url])
+            NotificationCenter.default.post(notification)
+        } else if isWeb {
+            let notification = Notification(name: .swifterCallback, object: nil, userInfo: [CallbackNotification.optionsURLKey: url])
+            NotificationCenter.default.post(notification)
+        }
+        return false
+    }
+    
     #endif
     
     public class func handleOpenURL(_ url: URL, callbackURL: URL) -> Bool {
```

Comment:
If the new callback method is supposed to replace `handleOpenURL(_:callbackUrl:)`, then please remove the old one, and supply a way to migrate to both. Like above, I don't think we should change the method signature of the existing one, but simply allow for SSO to be an additional argument that has a default value of `nil`.

Reply:
Fixed!
ba79795